picamera2/previews/drm_preview.py

```diff
@@ -63,7 +63,7 @@ class DrmPreview(NullPreview):
     def __init__(self, x=0, y=0, width=640, height=480, transform=None):
         self.init_drm(x, y, width, height, transform)
         self.stop_count = 0
-        self.fb = pykms.DumbFramebuffer(self.card, width, height, "AB24")
+        self.fb = pykms.DumbFramebuffer(self.card, width, height, "XR24")
```

Comment:
Just checking... presumably the display device understands these formats and won't cause a colour swap or anything?

Reply:
This is why I put in the commit text that it should really be asking the display what it supports.

RP1 supports XRGB8888, XBGR8888, RGB888, BGR888, and RGB565.
SPI displays typically only support XRGB8888 and RGB565.
Other displays could support other format combinations.

As I read it, this buffer is only used if decoding MJPEG, so it would be nicer if this was only allocated when needed, not on init. Presumably this means it needs to match the format that the MJPEG decoder is going to produce.

Buffer allocation is validated against the formats supported, hence why I've hacked it.

Reply:
Running on the Pi, I'm slightly surprised to find the both "XR24" and "XB24" work here, and there's no colour swap on the display. Do you think either of these is more likely to be supported on other devices?